source/app/controllers/urls_controller.rb

```diff
@@ -1,2 +1,42 @@
 class UrlsController < ApplicationController
+
+  def index
+    @urls = Url.all
+  end
+
+  def new
+  end
+
+  def create
+    @url = Url.new(url_params)
+
+    if @url.save
+      redirect_to '/'
+    else
+      @urls = Url.all
+      render 'index'
+    end
+  end
+
+  def destroy
+    @url = Url.find(params[:id])
+    @url.destroy
+
+    redirect_to '/'
+  end
+
+  def click_link
+    @url = Url.find(params[:format])
+
+    new_count = @url.click_count + 1
+    @url.update_attribute(:click_count, new_count)
```

Comment:
Yeah, sorry. I was just responding to your previous comment kinda out of context. It is super important to make this click counting work properly at web-scale when this bad boy gets to the front of Reddit. 😛